### ISSUE TITLE
Window Blind

### ISSUE THREAD

**Issue by Flawl3ssSWE** [jchnlemon/homebridge-telldus#43]
Hi I've got your code to work with a window blind. But there is some problems with it. For the first is besides it always spining like its going up or down when it's standing still and as second you cant set the procentage. It just ignores it and goes all the way down. Nice work btw. Hope you can fix these things!

**Comment by mifi** [comment]
IIRC this is a known problem, and because telldus does not report back the state of the blinds.
See discussion here:
https://github.com/jchnlemon/homebridge-telldus/issues/33

**Comment by Flawl3ssSWE** [comment]
@mifi ok but shouldnt it be able to do that if you get it to work with the stop button and if you write in a time for it to go down in the code. Could you fix the always spining wheel besides it?

**Comment by krfr** [comment]
My blinds don't report back status in Telldus. But @Flawl3ssSWE maybe have another devices?
I use rollertrol.

**Comment by Flawl3ssSWE** [comment]
@krfr me too (I have rollertrol). Quick question are you from Sweden and if did you order the blind from m.nu?

**Comment by krfr** [comment]
Yes, I'm from Sweden. I purchased my tube engines on eBay and not from m.nu. But it's same model I think, and maybe there is som difference that I don't aware of. 
I use Znet Lite V1 and in my case I can't get status from my blinds. But in V2 this maybe work?

**Comment by Flawl3ssSWE** [comment]
@krfr I just got the v2 today and it doesn't work

**Comment by krfr** [comment]
Then you probably have only up, down and stop as available commands in telldus.

In telldus you can't tell your blinds to close only, for example, to 50%, but you can hit stop button when you want. In homebridge and Siri there is no stop function for blinds at this moment. 

About your first question...
I'm not sure if I understand your question or problem correctly. Is it the "spinning progress symbol" in you home app that is spinning and your blinds doesn't move at all?

**Comment by Flawl3ssSWE** [comment]
Yes its not moving but the wheel by the side of it (in the Home app) is always spinning. @krfr @mifi

**Comment by krfr** [comment]
Okay. My spin too, but only for a short time and especially when I open the blind.
Since you do not receive confirmation from tube engine, Homebridge does not know when the end position has occurred.

**Comment by mifi** [comment]
I'm not sure if i understand the problem correctly either. Is the problem only that the spinning indicator doesn't go away? And is that really a problem apart from being a cosmetic issue?

**Comment by Flawl3ssSWE** [comment]
@mifi its always spinning its just in the app

**Comment by mifi** [comment]
You mean the home app?

**Comment by Flawl3ssSWE** [comment]
@mifi Yes. but can't you in the code add a section with time for up and for down time. So it can use that to know were it is?

**Comment by krfr** [comment]
What are your plans to use a function like this? Just curious.

**Comment by Flawl3ssSWE** [comment]
@krfr my plan is to use it in order to be able to set the % because if you use the down time you can calculate which postion it should have depending on the time

**Comment by krfr** [comment]
Okey. In my case I have four blinds, and there is differences in run-time from one engine to another. 
One engine take approximately 15 seconds to close, while another engine take 19 seconds.

But I still don't understand why. Don't you ever want to close your blinds entirely?

**Comment by Flawl3ssSWE** [comment]
@krfr yes I would like to close them entirely  at the night but in the afternoon the sun shines me right in my eyes when I'm at my pc. Then I would like to set it to like half way down.

**Comment by mifi** [comment]
Then it would have be configurable through the config file, approx how long it would take to open/close. Then we would have to model a virtual blinds device with an assumed state and timers simulating moving/stopping. I don't have time for this now but PR's are welcome. Guess it should be contained in its own node module with a simple api for setting percentage etc.

**Comment by Flawl3ssSWE** [comment]
@mifi That's exactly what I meant. But I'm not good at coding. When will you have time again?

**Comment by mifi** [comment]
I cannot give you any estimates. It's a hobby project for me, so it will also depend on my mood. And I don't have any blinds myself. But it's open source, so anyone is welcome to contribute.

**Comment by Flawl3ssSWE** [comment]
@mifi okay thanks for the help and for the plugin, love it!

**Comment by Flawl3ssSWE** [comment]
@mifi If you ever will do it I can help you test it. Thanks again
